server/data/tokenStore.ts

```diff
@@ -29,7 +29,7 @@ export default class TokenStore {
   }
 
   public async setToken(key: string, token: string, durationSeconds: number): Promise<void> {
-    this.setRedisAsync(key, token, 'EX', durationSeconds)
+    await this.setRedisAsync(key, token, 'EX', durationSeconds)
```

Comment:
I think you still need to `return` this promise or the client doesn't get it (it'd get a promise but not the one performing the set operation) - I _think_ TS doesn't complain only because the return value is `Promise<void>` and that probably is what an async function returns by default 🤷  (1)

Then the client awaits for this promise returned by this method to be fullfilled (client code seems to `await` already 👌 )

(1) if the signature of this function was something like `async function setToken(...): Promise<string>` it would not be happy by the fact you don't return a promise

Reply:
the callback for `redisClient.set` takes 2 params: an Error and the second is the constant `'OK'`. one of these will be `null`. `promisify` turns this into a promise which either throws the error or returns the constant `'OK'`. therefore, the resulting value of this promise is never interesting. presumably, that's why the `setToken` wrapper method promises a void response.

hence, i felt it wasn't necessary to return this internal/derived promise from inside the redis library. simply awaiting it means that the promise _we_ return is dependent on their one resolving.

alternatively, we could do:
```typescript
return this.setRedisAsync(key, token, 'EX', durationSeconds)
```
but then we'd need to change the type of `setToken` to have a return value of `Promise<'OK'>`. that'd be more true to what's happening in the redis library, but i suppose this method is intended to hide redis implementation details…?

Reply:
I think the Redis internals should be hidden yes.

I see, in this case add `await` means the method will wait for the promise to resolve and because we don't explicitly return any promise and this is an `async function`, it will return a `Promise<void>` which resolves instantly to `undefined` (but after the waiting to set the key in Redis) 👌 

Sorry, I see what's happening better now, I think this is a slightly special case because the it returns `Promise<void>` and we don't care about the actual promise resolved value.

Reply:
yeah exactly. i think quite often it'd be sensible to return the internal promise because you'd normally want the value inside it